runners/google-cloud-dataflow-java/worker/src/main/java/org/apache/beam/runners/dataflow/worker/streaming/Work.java

```diff
@@ -111,15 +128,58 @@ public Collection<Windmill.LatencyAttribution> getLatencyAttributions() {
       if (duration.equals(Duration.ZERO)) {
         continue;
       }
-      list.add(
-          Windmill.LatencyAttribution.newBuilder()
-              .setState(state)
-              .setTotalDurationMillis(duration.getMillis())
-              .build());
+      LatencyAttribution.Builder laBuilder = Windmill.LatencyAttribution.newBuilder();
+      if (state == LatencyAttribution.State.ACTIVE) {
+        laBuilder = addActiveLatencyBreakdownToBuilder(isHeartbeat, laBuilder, workId, sampler);
+      }
+      Windmill.LatencyAttribution la =
+          laBuilder.setState(state).setTotalDurationMillis(duration.getMillis()).build();
+      list.add(la);
     }
     return list;
   }
 
+  private static LatencyAttribution.Builder addActiveLatencyBreakdownToBuilder(
+      boolean isHeartbeat,
+      LatencyAttribution.Builder builder,
+      String workId,
+      DataflowExecutionStateSampler sampler) {
+    if (isHeartbeat) {
+      ActiveLatencyBreakdown.Builder stepBuilder = ActiveLatencyBreakdown.newBuilder();
+      ActiveMessageMetadata activeMessage = sampler.getActiveMessageMetadataForWorkId(workId);
```

Comment:
this returns Optional.  can use `!activeMessage.isPresent()`

Reply:
[075dd57](https://github.com/apache/beam/pull/29592/commits/075dd57cca2409b576222ea6eb014f3dfe368bf0)